optimizer/plan/planbuilder.go

```diff
@@ -121,9 +126,22 @@ func (b *planBuilder) extractSelectAgg(sel *ast.SelectStmt) []*ast.AggregateFunc
 				return nil
 			}
 			item.Expr = n.(ast.ExprNode)
+			// If item is PositionExpr, we need to rebind it.
+			// For PositionExpr will refer to a ResultField in fieldlist.
+			// After extract AggExpr from fieldlist, it may be changed (See the code above).
+			if pe, ok := item.Expr.(*ast.PositionExpr); ok {
+				pe.Refer = sel.GetResultFields()[pe.N-1]
+			}
+		}
+	}
+	if sel.Having != nil {
```

Comment:
`having` is the process before `order by`, so I think it's better to extract from `having` before `order by`.